internal/venice-client-common/src/main/java/com/linkedin/venice/compression/GzipCompressor.java

```diff
@@ -95,6 +95,22 @@ public ByteBuffer decompress(byte[] data, int offset, int length) throws IOExcep
     }
   }
 
+  @Override
+  public ByteBuffer decompressAndPrependSchemaHeader(byte[] data, int length) throws IOException {
+    int valueSchema = ByteUtils.readInt(data, 0);
+    byte[] decompressedByteArray;
+    try (InputStream gis =
+        decompress(new ByteArrayInputStream(data, SCHEMA_HEADER_LENGTH, length - SCHEMA_HEADER_LENGTH))) {
+      decompressedByteArray = IOUtils.toByteArray(gis);
+      ByteBuffer byteBufferWithHeader = ByteBuffer.allocate(SCHEMA_HEADER_LENGTH + decompressedByteArray.length)
+          .putInt(valueSchema)
+          .put(decompressedByteArray);
+      byteBufferWithHeader.flip();
```

Comment:
Nitpick: `flip()` will do:

```
    public Buffer flip() {
        limit = position;
        position = 0;
        mark = -1;
        return this;
    }
```

But it seems like we only care about the `limit = position` part of that. I think it's more clear to do just the operation we need, than to do this package of operations, e.g.:

```
byteBufferWithHeader.limit(byteBufferWithHeader.position());
byteBufferWithHeader.position(SCHEMA_HEADER_LENGTH);
```

Reply:
Sure, thanks for pointing out, I can update this.